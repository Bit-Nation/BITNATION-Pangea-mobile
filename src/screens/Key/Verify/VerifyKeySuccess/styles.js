import { MediaQueryStyleSheet } from 'react-native-responsive';
import Colors from '../../../../global/Colors';

const styles = MediaQueryStyleSheet.create(
  {
    container: {
      flexDirection: 'column',
      flex: 1,
      paddingTop: 50,

    },
    backgroundImage: {
      width: '100%',
      position: 'absolute',
    },

    header: {
      backgroundColor: Colors.Transparent,
    },
    cancel: {
      color: Colors.Blue
    },
    next: {
      color: Colors.Blue,
    },
    title: {
      color: Colors.Background,
      fontSize: 26,
      fontWeight: '400',
      marginLeft: '3%',
      backgroundColor: Colors.Transparent,
    },
    wholeText: {
      margin: '3%',
      marginTop: '16%',
      marginRight: '2%'
    },
    applyBold:{
      fontWeight: 'bold',
    },
    description: {
      color: Colors.Blue,
      fontSize: 16,
      marginBottom: '6%',
      backgroundColor: Colors.Transparent,
    },
    buttonContainer: {
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: '80%',
    },
  });
export default styles;
