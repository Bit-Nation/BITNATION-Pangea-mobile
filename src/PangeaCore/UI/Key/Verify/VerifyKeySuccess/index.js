// @flow

import React from 'react';
import { View } from 'react-native';
import styles from './styles';
import BackgroundImage from 'pangea-common-reactnative/UI/BackgroundImage';
import FakeNavigationBar from 'pangea-common-reactnative/UI/FakeNavigationBar';
import BodyParagraphs from 'pangea-common-reactnative/UI/BodyParagraphs';
import i18n from 'pangea-common/i18n';
import PanelView from 'pangea-common-reactnative/UI/PanelView';
import NavigatorComponent from 'pangea-common-reactnative/UI/NavigatorComponent';

class VerifyKeySuccess extends NavigatorComponent<void> {
  static navigatorButtons = { leftButtons: [], rightButtons: [] };

  onNextButtonPressed = () => {
    if (this.props.navigator) {
      this.props.navigator.popToRoot();
    }
  };

  render() {
    return (
      <View style={styles.screenContainer}>
        <BackgroundImage />
        <FakeNavigationBar />

        <View style={styles.bodyContainer}>
          <PanelView
            style={styles.panelViewTransparent}
            childrenContainerStyle={styles.noflex}
            buttonTitle={i18n.t('screens.verifyKey.success.doneButton')}
            onButtonClick={this.onNextButtonPressed}
          >
            <BodyParagraphs paragraphs={i18n.t('screens.verifyKey.success.instructions')} />
          </PanelView>

        </View>
      </View>
    );
  }
}

export default VerifyKeySuccess;
