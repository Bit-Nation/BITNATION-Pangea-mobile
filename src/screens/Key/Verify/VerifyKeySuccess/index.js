// @flow

import React from 'react';
import { View } from 'react-native';
import styles from './styles';
import BackgroundImage from '../../../../components/common/BackgroundImage';
import FakeNavigationBar from '../../../../components/common/FakeNavigationBar';
import BodyParagraphs from '../../../../components/common/BodyParagraphs';
import i18n from '../../../../global/i18n';
import PanelView from '../../../../components/common/PanelView';
import NavigatorComponent from '../../../../components/common/NavigatorComponent';

class VerifyKeySuccess extends NavigatorComponent<void> {
  static navigatorButtons = { leftButtons: [], rightButtons: [] };

  onNextButtonPressed = () => {
    if (this.props.navigator) {
      this.props.navigator.popToRoot();
    }
  };

  render() {
    return (
      <View style={styles.screenContainer}>
        <BackgroundImage />
        <FakeNavigationBar />

        <View style={styles.bodyContainer}>
          <PanelView
            style={styles.panelViewTransparent}
            childrenContainerStyle={styles.noflex}
            buttonTitle={i18n.t('screens.verifyKey.success.doneButton')}
            onButtonClick={this.onNextButtonPressed}
          >
            <BodyParagraphs paragraphs={i18n.t('screens.verifyKey.success.instructions')} />
          </PanelView>

        </View>
      </View>
    );
  }
}

export default VerifyKeySuccess;
