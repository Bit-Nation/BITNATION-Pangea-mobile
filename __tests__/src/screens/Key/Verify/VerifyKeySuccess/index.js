import React from 'react';
import { shallow } from 'enzyme';

import VerifyKeySuccess from '../../../../../../src/screens/Key/Verify/VerifyKeySuccess';
import navigatorMock from '../../../../../../__mocks__/Navigator';

test('VerifyKeySuccess renders correctly', () => {
  const propsMock = {
    navigator: navigatorMock,
  };

  const wrapper = shallow((
    <VerifyKeySuccess {...propsMock} />
  ));
  expect(wrapper).toMatchSnapshot();
});
