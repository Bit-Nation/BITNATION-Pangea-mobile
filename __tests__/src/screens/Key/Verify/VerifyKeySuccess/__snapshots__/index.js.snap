// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyKeySuccess renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyKeySuccess
    navigator={
      Object {
        "push": [MockFunction],
        "setOnNavigatorEvent": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BackgroundImage
          source={
            Object {
              "testUri": "../../../src/assets/images/background.jpg",
            }
          }
          style={undefined}
        />,
        <FakeNavigationBar
          navBarHidden={false}
        />,
        <View
          style={
            Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            }
          }
        >
          <PanelView
            buttonTitle="Done"
            childrenContainerStyle={
              Object {
                "flex": 0,
              }
            }
            onButtonClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            titleStyle={
              Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          >
            <BodyParagraphs
              paragraphs={
                Array [
                  "Congratulations! You correctly entered your private key.",
                  "Now, put your paper with your private key in a safe place.",
                ]
              }
            />
          </PanelView>
        </View>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": Object {
            "testUri": "../../../src/assets/images/background.jpg",
          },
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navBarHidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <PanelView
            buttonTitle="Done"
            childrenContainerStyle={
              Object {
                "flex": 0,
              }
            }
            onButtonClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            titleStyle={
              Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          >
            <BodyParagraphs
              paragraphs={
                Array [
                  "Congratulations! You correctly entered your private key.",
                  "Now, put your paper with your private key in a safe place.",
                ]
              }
            />
          </PanelView>,
          "style": Object {
            "alignContent": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginLeft": 8,
            "marginRight": 8,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonTitle": "Done",
            "children": <BodyParagraphs
              paragraphs={
                Array [
                  "Congratulations! You correctly entered your private key.",
                  "Now, put your paper with your private key in a safe place.",
                ]
              }
            />,
            "childrenContainerStyle": Object {
              "flex": 0,
            },
            "onButtonClick": [Function],
            "style": Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginBottom": 8,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 8,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            "titleStyle": Object {
              "color": "white",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "paragraphs": Array [
                "Congratulations! You correctly entered your private key.",
                "Now, put your paper with your private key in a safe place.",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BackgroundImage
            source={
              Object {
                "testUri": "../../../src/assets/images/background.jpg",
              }
            }
            style={undefined}
          />,
          <FakeNavigationBar
            navBarHidden={false}
          />,
          <View
            style={
              Object {
                "alignContent": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "marginLeft": 8,
                "marginRight": 8,
              }
            }
          >
            <PanelView
              buttonTitle="Done"
              childrenContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              onButtonClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Congratulations! You correctly entered your private key.",
                    "Now, put your paper with your private key in a safe place.",
                  ]
                }
              />
            </PanelView>
          </View>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": Object {
              "testUri": "../../../src/assets/images/background.jpg",
            },
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navBarHidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <PanelView
              buttonTitle="Done"
              childrenContainerStyle={
                Object {
                  "flex": 0,
                }
              }
              onButtonClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              titleStyle={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            >
              <BodyParagraphs
                paragraphs={
                  Array [
                    "Congratulations! You correctly entered your private key.",
                    "Now, put your paper with your private key in a safe place.",
                  ]
                }
              />
            </PanelView>,
            "style": Object {
              "alignContent": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-start",
              "marginLeft": 8,
              "marginRight": 8,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonTitle": "Done",
              "children": <BodyParagraphs
                paragraphs={
                  Array [
                    "Congratulations! You correctly entered your private key.",
                    "Now, put your paper with your private key in a safe place.",
                  ]
                }
              />,
              "childrenContainerStyle": Object {
                "flex": 0,
              },
              "onButtonClick": [Function],
              "style": Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              "titleStyle": Object {
                "color": "white",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "paragraphs": Array [
                  "Congratulations! You correctly entered your private key.",
                  "Now, put your paper with your private key in a safe place.",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
